content/source/assets/javascripts/application.js

```diff
@@ -31,28 +37,94 @@ document.addEventListener("turbolinks:load", function() {
         e.preventDefault();
         $(this).parent("li").trigger("click");
     });
-    // Navs can include an optional global toggle like this:
-    // <a href="#" class="subnav-toggle">(Expand/collapse all)</a>
-    // Navs that don't want that can leave it out.
-    var globalExpand = true;
-    $("#docs-sidebar a.subnav-toggle").on("click", function(e) {
-        e.preventDefault();
-        if (globalExpand) {
-            subNavs.addClass("active");
-        } else {
-            subNavs.removeClass("active");
+
+    // If this is a Very Large Sidebar, add extra controls to expand/collapse
+    // and filter it.
+    var sidebarLinks = docsSidebar.find("a");
+    if (sidebarLinks.length > 30) {
+        if ($("#sidebar-controls").length === 0) { // then add it!
+            var sidebarControlsHTML =
+                '<div id="sidebar-controls">' +
+                    '<label for="sidebar-filter-field">Filter page titles</label>' +
```

Comment:
Should this be `Filter navigation` or `Filter sidebar` to be a little bit more clear what its doing? In my testing it seems to handle navigation headers just fine. 👍 

```suggestion
                    '<label for="sidebar-filter-field">Filter navigation</label>' +
```